projects/components/src/action-menu/action-menu.component.ts

```diff
@@ -0,0 +1,194 @@
+        return actions.filter(
+            action => !action.availability || action.availability(selection) || this.isActionDisabled(action)
+        );
+    }
+
+    /**
+     * List of only the actions that are marked as {@link ActionType.STATIC}
+     */
+    get staticActions(): ActionItem<R, T>[] {
+        const staticActions = this.actions.filter(action => action.actionType === ActionType.STATIC);
+        return this.getAvailableActions(staticActions);
+    }
+
+    /**
+     * List of only the actions that are marked as {@link ActionType.STATIC_FEATURED}
+     */
+    get staticFeaturedActions(): ActionItem<R, T>[] {
+        const staticActions = this.actions.filter(action => action.actionType === ActionType.STATIC_FEATURED);
+        return this.getAvailableActions(staticActions);
+    }
+
+    private getFlattenedActionList(actions: ActionItem<R, T>[], actionType: ActionType): ActionItem<R, T>[] {
+        let featuredActions: ActionItem<R, T>[] = [];
+        actions.forEach(action => {
+            if (action.children && action.children.length) {
+                featuredActions = featuredActions.concat(this.getFlattenedActionList(action.children, actionType));
+            } else if (action.actionType === actionType) {
+                featuredActions.push(action);
+            }
+        });
+        return featuredActions;
+    }
+
+    /**
+     * Actions that depend on selected entities and belong to main menu list. The returned list length is less than the
+     * configured featured count in {@link actionDisplayConfig}
+     * @param selection The selected entities based on which the actions availability is calculated
+     */
+    getContextualFeaturedActions(selection: R[]): ActionItem<R, T>[] {
+        const flattenedFeaturedActionList = this.getFlattenedActionList(this.actions, ActionType.CONTEXTUAL_FEATURED);
+        const availableFeaturedActions = this.getAvailableActions(flattenedFeaturedActionList, selection);
+        return this.actionDisplayConfig.contextual.featuredCount
+            ? availableFeaturedActions.slice(0, this.actionDisplayConfig.contextual.featuredCount)
+            : availableFeaturedActions;
+    }
+
+    /**
+     * Actions that depend on selected entities but belong to sub menu
+     * @param selection The selected entities based on which the actions availability is calculated
+     */
+    getContextualActions(selection?: R[]): ActionItem<R, T>[] {
+        const contextualActions = this.actions.filter(
+            action => action.actionType !== ActionType.STATIC_FEATURED && action.actionType !== ActionType.STATIC
+        );
+        return this.getAvailableActions(contextualActions, selection);
+    }
+
+    /**
+     * Execute the {@link ActionItem.handler} function by passing it {@link #selectedEntities} and
+     * {@link ActionItem.handlerData} as arguments
+     */
+    runActionHandler(action: ActionItem<R, T>): void {
+        const actionHandlerData = [];
```

Comment:
Should just call

```ts
action.handler(this.selectedEntities, action.handlerData1)
```

RIght now you're breaking the signature if there are no selected entities